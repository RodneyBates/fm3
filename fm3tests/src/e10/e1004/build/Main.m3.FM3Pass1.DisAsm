 ItkEOF()
 ItkRightEnd()
 ItkModuleLtPatch(C4,L1(16_1),[1,0])
 ItkModuleBeginPatch(C669,L1(16_1),[29,2])
 ItkModuleRt(L1(16_1),[1,0])
 ItkDeclScopeRt(L1(16_1))
 ItkOpenScopeRt(L1(16_1))
 ItkStmtListRt(L0(16_0),[30,2])
 ItkStmtListLt(L0(16_0),[30,2])
 ItkOpenDeclListLtPatch(C10,L3(16_3),[7,2])
 ItkOpenDeclListRt(L3(16_3),[7,2])
 ItkProcDefLtPatch(C322,Id8"P",[20,2])
 ItkProcDefRt(Id13"ppp",[27,8])
 ItkOpenScopeRt(L4(16_4))
 ItkDeclScopeRt(L5(16_5))
 ItkOpenScopeRt(L5(16_5))
 ItkStmtListRt(L0(16_0),[27,4])
 ItkStmtListLt(L0(16_0),[27,4])
 ItkOpenDeclListRt(L0(16_0),[26,4])
 ItkOpenDeclListLt(L0(16_0),[26,4])
 ItkOpenScopeLt(L5(16_5))
 ItkDeclScopeLt(L5(16_5))
 ItkOpenScopeLt(L4(16_4))
 ItkDeclScopeRt(L4(16_4))
 ItkProcDefBody([26,2])
 ItkSignatureProperLtPatch(C340,[21,4])
 ItkSignatureProperRt([21,4])
 ItkRaisesSetRt(L0(16_0),[26,2])
 ItkRaisesSetLt(L0(16_0),[26,2])
 ItkResultTypeAbsent([26,2])
 ItkFormalsListLtPatch(C340,L4(16_4),[21,6])
 ItkFormalsListRt(L4(16_4),[21,6])
 ItkFormalsListSepPatch(C507,L3(16_3),[24,4])
 ItkSkipLtPatch(C507,L7(16_7))
 ItkSkipRt(L7(16_7))
 ItkDeclValAbsent([25,4])
 ItkROFormalValue([25,4])
 ItkDeclTypeAbsent([24,6])
 ItkROFormalType([25,4])
 ItkROFormalIdListLtPatch(C507,L1(16_1),[24,15])
 ItkROFormalIdListRt(L1(16_1),[24,15])
 ItkDeclId(L13(16_d),Id12"PF4",L-9223372036854775808(16_8000000000000000),[24,15])
 ItkFormalsListSepPatch(C449,L2(16_2),[23,4])
 ItkSkipLtPatch(C449,L6(16_6))
 ItkSkipRt(L6(16_6))
 ItkDeclValAbsent([24,4])
 ItkVARFormalValue([24,4])
 ItkDeclTypeAbsent([23,6])
 ItkVARFormalType([24,4])
 ItkVARFormalIdListLtPatch(C449,L1(16_1),[23,10])
 ItkVARFormalIdListRt(L1(16_1),[23,10])
 ItkDeclId(L12(16_c),Id11"PF3",L-9223372036854775808(16_8000000000000000),[23,10])
 ItkFormalsListSepPatch(C391,L1(16_1),[22,4])
 ItkSkipLtPatch(C391,L5(16_5))
 ItkSkipRt(L5(16_5))
 ItkDeclValAbsent([23,4])
 ItkVALUEFormalValue([23,4])
 ItkDeclTypeAbsent([22,6])
 ItkVALUEFormalType([23,4])
 ItkVALUEFormalIdListLtPatch(C391,L1(16_1),[22,12])
 ItkVALUEFormalIdListRt(L1(16_1),[22,12])
 ItkDeclId(L11(16_b),Id10"PF2",L-9223372036854775808(16_8000000000000000),[22,12])
 ItkSkipLtPatch(C340,L4(16_4))
 ItkSkipRt(L4(16_4))
 ItkDeclValAbsent([22,4])
 ItkVALUEFormalValue([22,4])
 ItkDeclTypeAbsent([21,6])
 ItkVALUEFormalType([22,4])
 ItkVALUEFormalIdListLtPatch(C340,L1(16_1),[21,6])
 ItkVALUEFormalIdListRt(L1(16_1),[21,6])
 ItkDeclId(L11(16_b),Id9"PF1",L-9223372036854775808(16_8000000000000000),[21,6])
 ItkDeclScopeLt(L4(16_4))
 ItkDeclId(L18(16_12),Id8"P",L-9223372036854775808(16_8000000000000000),[20,12])
 ItkOpenDeclListSep(L2(16_2),[7,2])
 ItkTypeDeclLtPatch(C175,[15,7])
 ItkTypeDeclTypePatch(C190,[16,4])
 ItkTypeDeclRt([15,7])
 ItkDeclScopeRt(L3(16_3))
 ItkObjTypeLtPatch(C190,L0(16_0),L3(16_3),[16,6])
 ItkObjTypeRt(L0(16_0),L3(16_3),[16,6])
 ItkOverrideListRt(L0(16_0),[18,6])
 ItkOverrideListLt(L0(16_0),[18,6])
 ItkMethodDeclListRt(L0(16_0),[18,6])
 ItkMethodDeclListLt(L0(16_0),[18,6])
 ItkFieldDeclListLtPatch(C201,L0(16_0),[17,8])
 ItkFieldDeclListRt(L0(16_0),[17,8])
 ItkSkipLtPatch(C228,L3(16_3))
 ItkSkipRt(L3(16_3))
 ItkDeclValAbsent([18,6])
 ItkFieldDeclValue([18,6])
 ItkDeclTypeAbsent([18,6])
 ItkFieldDeclType([17,8])
 ItkFieldDeclIdListLtPatch(C201,L1(16_1),[17,8])
 ItkFieldDeclIdListRt(L1(16_1),[17,8])
 ItkDeclId(L16(16_10),Id7"OOOOF1",L-9223372036854775808(16_8000000000000000),[17,8])
 ItkDeclScopeLt(L3(16_3))
 ItkBrandAbsent([16,6])
 ItkSupertypeAbsent([16,6])
 ItkDeclId(L8(16_8),Id6"O",L-9223372036854775808(16_8000000000000000),[15,7])
 ItkOpenDeclListSep(L1(16_1),[7,2])
 ItkTypeDeclLtPatch(C59,[9,7])
 ItkTypeDeclTypePatch(C74,[10,4])
 ItkTypeDeclRt([9,7])
 ItkRecTypeLtPatch(C74,L0(16_0),[10,6])
 ItkRecTypeRt(L0(16_0),[10,6])
 ItkDeclScopeRt(L2(16_2))
 ItkFieldDeclListLtPatch(C77,L0(16_0),[11,8])
 ItkFieldDeclListRt(L0(16_0),[11,8])
 ItkSkipLtPatch(C104,L2(16_2))
 ItkSkipRt(L2(16_2))
 ItkDeclValAbsent([12,6])
 ItkFieldDeclValue([12,6])
 ItkDeclTypeAbsent([12,6])
 ItkFieldDeclType([11,8])
 ItkFieldDeclIdListLtPatch(C77,L1(16_1),[11,8])
 ItkFieldDeclIdListRt(L1(16_1),[11,8])
 ItkDeclId(L15(16_f),Id5"RF1",L-9223372036854775808(16_8000000000000000),[11,8])
 ItkDeclScopeLt(L2(16_2))
 ItkDeclId(L8(16_8),Id4"R",L-9223372036854775808(16_8000000000000000),[9,7])
 ItkSkipLtPatch(C36,L1(16_1))
 ItkSkipRt(L1(16_1))
 ItkDeclValAbsent([9,0])
 ItkVarDeclValue([9,0])
 ItkDeclTypeAbsent([9,0])
 ItkVarDeclType([7,6])
 ItkVarDeclIdListLtPatch(C10,L1(16_1),[7,6])
 ItkVarDeclIdListRt(L1(16_1),[7,6])
 ItkDeclId(L10(16_a),Id3"V",L-9223372036854775808(16_8000000000000000),[7,6])
 ItkOpenScopeLt(L1(16_1))
 ItkDeclScopeLt(L1(16_1))
 ItkLeftEnd()
 ItkBOF()
     0
