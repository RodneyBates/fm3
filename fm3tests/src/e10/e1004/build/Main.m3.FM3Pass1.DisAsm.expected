 ItkEOF()
 ItkRightEnd()
 ItkModuleLtPatch(C4,L2(16_2),[1,0])
 ItkModuleBeginPatch(C665,L2(16_2),[29,2])
 ItkModuleRt(L2(16_2),[1,0])
 ItkDeclScopeRt(L1(16_1))
 ItkOpenScopeRt(L1(16_1))
 ItkStmtListRt(L0(16_0),[30,2])
 ItkStmtListLt(L0(16_0),[30,2])
 ItkOpenDeclListLtPatch(C10,L3(16_3),[7,2])
 ItkOpenDeclListRt(L3(16_3),[7,2])
 ItkProcDefLtPatch(C318,Id8"P",[20,2])
 ItkProcDefRt(Id13"ppp",[20,2])
 ItkDeclScopeRt(L4(16_4))
 ItkOpenScopeRt(L4(16_4))
 ItkDeclScopeRt(L5(16_5))
 ItkOpenScopeRt(L5(16_5))
 ItkStmtListRt(L0(16_0),[27,4])
 ItkStmtListLt(L0(16_0),[27,4])
 ItkOpenDeclListRt(L0(16_0),[26,4])
 ItkOpenDeclListLt(L0(16_0),[26,4])
 ItkOpenScopeLt(L5(16_5))
 ItkDeclScopeLt(L5(16_5))
 ItkOpenScopeLt(L4(16_4))
 ItkProcDefBody([26,2])
 ItkDeclId(L18(16_12),Id8"P",L-9223372036854775808(16_8000000000000000),[20,12])
 ItkSignatureProperLtPatch(C321,[21,4])
 ItkSignatureProperRt([21,4])
 ItkRaisesSetRt(L0(16_0),[26,2])
 ItkRaisesSetLt(L0(16_0),[26,2])
 ItkResultTypeAbsent([26,2])
 ItkFormalsListLtPatch(C321,L4(16_4),[21,6])
 ItkFormalsListRt(L4(16_4),[21,6])
 ItkFormalsListSepPatch(C488,L3(16_3),[24,4])
 ItkSkipLtPatch(C488,L7(16_7))
 ItkSkipRt(L7(16_7))
 ItkDeclValAbsent([24,6])
 ItkROFormalValue([25,4])
 ItkDeclTypeAbsent([24,6])
 ItkROFormalType([25,4])
 ItkROFormalIdListLtPatch(C488,L1(16_1),[24,15])
 ItkROFormalIdListRt(L1(16_1),[24,15])
 ItkDeclId(L13(16_d),Id12"PF4",L-9223372036854775808(16_8000000000000000),[24,15])
 ItkFormalsListSepPatch(C430,L2(16_2),[23,4])
 ItkSkipLtPatch(C430,L6(16_6))
 ItkSkipRt(L6(16_6))
 ItkDeclValAbsent([23,6])
 ItkVARFormalValue([24,4])
 ItkDeclTypeAbsent([23,6])
 ItkVARFormalType([24,4])
 ItkVARFormalIdListLtPatch(C430,L1(16_1),[23,10])
 ItkVARFormalIdListRt(L1(16_1),[23,10])
 ItkDeclId(L12(16_c),Id11"PF3",L-9223372036854775808(16_8000000000000000),[23,10])
 ItkFormalsListSepPatch(C372,L1(16_1),[22,4])
 ItkSkipLtPatch(C372,L5(16_5))
 ItkSkipRt(L5(16_5))
 ItkDeclValAbsent([22,6])
 ItkVALUEFormalValue([23,4])
 ItkDeclTypeAbsent([22,6])
 ItkVALUEFormalType([23,4])
 ItkVALUEFormalIdListLtPatch(C372,L1(16_1),[22,12])
 ItkVALUEFormalIdListRt(L1(16_1),[22,12])
 ItkDeclId(L11(16_b),Id10"PF2",L-9223372036854775808(16_8000000000000000),[22,12])
 ItkSkipLtPatch(C321,L4(16_4))
 ItkSkipRt(L4(16_4))
 ItkDeclValAbsent([21,6])
 ItkVALUEFormalValue([22,4])
 ItkDeclTypeAbsent([21,6])
 ItkVALUEFormalType([22,4])
 ItkVALUEFormalIdListLtPatch(C321,L1(16_1),[21,6])
 ItkVALUEFormalIdListRt(L1(16_1),[21,6])
 ItkDeclId(L11(16_b),Id9"PF1",L-9223372036854775808(16_8000000000000000),[21,6])
 ItkDeclScopeLt(L4(16_4))
 ItkOpenDeclListSep(L2(16_2),[7,2])
 ItkTypeDeclLtPatch(C175,[15,7])
 ItkTypeDeclTypePatch(C190,[16,4])
 ItkTypeDeclRt([15,7])
 ItkObjTypeLtPatch(C190,[16,6])
 ItkObjTypeRt([16,6])
 ItkDeclScopeRt(L3(16_3))
 ItkOverrideListRt(L0(16_0),[18,6])
 ItkOverrideListLt(L0(16_0),[18,6])
 ItkMethodDeclListRt(L0(16_0),[18,6])
 ItkMethodDeclListLt(L0(16_0),[18,6])
 ItkFieldDeclListLtPatch(C201,L0(16_0),[17,8])
 ItkFieldDeclListRt(L0(16_0),[17,8])
 ItkSkipLtPatch(C201,L3(16_3))
 ItkSkipRt(L3(16_3))
 ItkDeclValAbsent([18,6])
 ItkFieldDeclValue([18,6])
 ItkDeclTypeAbsent([18,6])
 ItkFieldDeclType([17,8])
 ItkFieldDeclIdListLtPatch(C201,L1(16_1),[17,8])
 ItkFieldDeclIdListRt(L1(16_1),[17,8])
 ItkDeclId(L16(16_10),Id7"OOOOF1",L-9223372036854775808(16_8000000000000000),[17,8])
 ItkDeclScopeLt(L3(16_3))
 ItkBrandAbsent([16,6])
 ItkSupertypeAbsent([16,6])
 ItkDeclId(L8(16_8),Id6"O",L-9223372036854775808(16_8000000000000000),[15,7])
 ItkOpenDeclListSep(L1(16_1),[7,2])
 ItkTypeDeclLtPatch(C59,[9,7])
 ItkTypeDeclTypePatch(C74,[10,4])
 ItkTypeDeclRt([9,7])
 ItkRecTypeLtPatch(C74,L0(16_0),[10,6])
 ItkRecTypeRt(L0(16_0),[10,6])
 ItkDeclScopeRt(L2(16_2))
 ItkFieldDeclListLtPatch(C77,L0(16_0),[11,8])
 ItkFieldDeclListRt(L0(16_0),[11,8])
 ItkSkipLtPatch(C77,L2(16_2))
 ItkSkipRt(L2(16_2))
 ItkDeclValAbsent([12,6])
 ItkFieldDeclValue([12,6])
 ItkDeclTypeAbsent([12,6])
 ItkFieldDeclType([11,8])
 ItkFieldDeclIdListLtPatch(C77,L1(16_1),[11,8])
 ItkFieldDeclIdListRt(L1(16_1),[11,8])
 ItkDeclId(L15(16_f),Id5"RF1",L-9223372036854775808(16_8000000000000000),[11,8])
 ItkDeclScopeLt(L2(16_2))
 ItkDeclId(L8(16_8),Id4"R",L-9223372036854775808(16_8000000000000000),[9,7])
 ItkSkipLtPatch(C10,L1(16_1))
 ItkSkipRt(L1(16_1))
 ItkDeclValAbsent([9,0])
 ItkVarDeclValue([9,0])
 ItkDeclTypeAbsent([9,0])
 ItkVarDeclType([7,6])
 ItkVarDeclIdListLtPatch(C10,L1(16_1),[7,6])
 ItkVarDeclIdListRt(L1(16_1),[7,6])
 ItkDeclId(L10(16_a),Id3"V",L-9223372036854775808(16_8000000000000000),[7,6])
 ItkOpenScopeLt(L1(16_1))
 ItkDeclScopeLt(L1(16_1))
 ItkLeftEnd()
 ItkBOF()
     0
